docs/developer/testing.md

```diff
@@ -0,0 +1,75 @@
+## Running unit tests using CLI
+
+To run unit tests from command line use following command (being in btp-manager main directory): 
+
+```shell
+make test
+```
+Details concerning `test` rule (prerequisites and recipe) are defined in `./Makefile`.
+
+By default, unit tests are run using `env-test` cluster. 
+Part of unit tests is implemented using [gingko](https://onsi.github.io/ginkgo/#top) library but all unit tests are invoked using `go test ./... <some options`
+You can find exact invocation reflected in the console output along with messages confirming whether `env-test` is being used.
+```
+STEP: bootstrapping test environment @ 01/13/23 08:24:45.981
+2023-01-13T08:24:45.981+0100  DEBUG   controller-runtime.test-env     starting control plane
+```
+
+### Running unit tests on existing cluster
+
+You can run tests on existing cluster (not `env-test`) executing helper script:
+
+```shell
+testing/run-unit-tests-on-cluster.sh
+```
+
+Script sets environment variables accordingly and invokes the same `make` target - i.e. `test`. 
+Ginkgo unit tests will use cluster in the current context, name of the cluster can be found in console output e.g.:
+```
+Running unit tests on the existing "k3d-test-cluster" cluster
+```
+> **NOTE:** Script attempts to restore the proper state of the cluster before each test run by removing resources that could be left if tests failed to properly remove those.
+> If problems with left-over resources recur, you can recreate the cluster.
+
+### Test output verbosity
+
+Setting for the `go test` verbosity is `-v` (verbose, print the full output event for passing tests). This can be changed in `make` recipe. 
+For `ginkgo` tests execution default setting is `-ginkgo.v` (verbose) and can be changed e.g. for `very verbose` by setting environment variable:
+```shell
+export GINKGO_VERBOSE_FLAG=-ginkgo.vv
+testing/run-unit-tests-on-cluster.sh
+```
+Alternatively it can be changed in `./Makefile` e.g.
+```Makefile
+GINKGO_VERBOSE_FLAG ?= -ginkgo.vv
```

Comment:
when moved to set-env-vars.sh this is not needed. All testing envs will be placed in the set-env-test.sh

Reply:
Changes pushed.###